internal/renderer/yatee/driver_test.go

```diff
@@ -0,0 +1,64 @@
+// +build experimental
+
+package yatee
+
+import (
+	"testing"
+
+	"gotest.tools/assert"
+	is "gotest.tools/assert/cmp"
+)
+
+var (
+	compose = `version: "3.4"
+services:
+  "@if $myapp.enable":
+    enabledservice:
+      image: alpine:$myapp.alpine_version
+      command: top $$foo
+  other:
+    image: nginx`
+	settings = map[string]interface{}{
+		"myapp": map[string]interface{}{
+			"enable":         true,
+			"alpine_version": "3.7",
+		},
+	}
+	expectedCompose = `services:
+  enabledservice:
+    command: top $$foo
+    image: alpine:3.7
+  other:
+    image: nginx
+version: "3.4"
+`
+)
+
+func TestDriverErrors(t *testing.T) {
+	testCases := []struct {
+		name          string
+		template      string
+		settings      map[string]interface{}
+		expectedError string
+	}{
+		{
+			name:          "no settings",
+			template:      "service: $loop",
+			expectedError: "variable 'loop' not set",
+		},
+	}
+	d := &Driver{}
+	for _, tc := range testCases {
```

Comment:
Is this a loop on a constant slice with one element?

Reply:
kinda :sweat_smile: